FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/TextFacetHandlerTest.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+namespace FinSearchUnified\Tests\Bundle\SearchBundleFindologic\FacetHandler;
+
+use FinSearchUnified\Bundle\SearchBundleFindologic\FacetHandler\TextFacetHandler;
+use Shopware\Bundle\SearchBundle\Condition\ProductAttributeCondition;
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\SearchBundle\FacetResult\RadioFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListItem;
+use Shopware\Components\Test\Plugin\TestCase;
+use SimpleXMLElement;
+
+class TextFacetHandlerTest extends TestCase
+{
+    /**
+     * @dataProvider filterProvider
+     *
+     * @param string $name
+     * @param string $type
+     * @param bool $doesSupport
+     */
+    public function testSupportsFilter($name, $type, $doesSupport)
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $filter->addChild('name', $name);
+        $filter->addChild('type', $type);
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->supportsFilter($filter);
+
+        $this->assertSame($doesSupport, $result);
+    }
+
+    public function filterProvider()
+    {
+        return [
+            'Category filter with "select" type' => ['cat', 'select', false],
+            'Category filter with "label" type' => ['cat', 'label', false],
+            'Vendor filter with "select" type' => ['vendor', 'label', true],
+            'Vendor filter with "label" type' => ['vendor', 'label', true],
+        ];
+    }
+
+    public function testFacetModeForBooleanResult()
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $facet = new ProductAttributeFacet(
+            'field',
+            ProductAttributeFacet::MODE_BOOLEAN_RESULT,
+            'field',
+            'Field'
+        );
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, new Criteria(), $filter);
+
+        $this->assertNull($result);
+    }
+
+    /**
+     * @dataProvider valueListDataProvider
+     *
+     * @param array $filterData
+     * @param ValueListFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForListFacetResult(
+        array $filterData,
+        ValueListFacetResult $facetResult,
+        ConditionInterface $condition = null
```

Comment:
Do you know why this works? `null` is definitely not an instance of `ConditionInterface`, so shouldn't this cause an fatal error?

Reply:
it works because we are providing a default value to the interface. If we don't provide the default value and then pass `null` from the data provider, it would throw an error. There is a feature in I think php 7 which typehints a null value but it is not supported in 5.6 so I used the default value instead.